kibana/idsnamesmapping/mapping.py

```diff
@@ -0,0 +1,132 @@
+"""
+Script for retrieving the mapping of flow ids to flow names and nodes ids to node titles.
+"""
+
+import argparse
+from datetime import datetime
+from hashlib import sha256
+import requests
+from opensearchpy import OpenSearch, helpers
+
+ANALYTICS_INDEX_PATTERN = "dev-analytics"
+
+
+def create_bulk_index_action(index_to_update: str, document_id: str, document: dict):
+    """
+    Creates an individual Bulk API index action.
+    """
+    return {
+        "_index": index_to_update,
+        "_type": "_doc",
+        "_id": document_id,
+        "_source": document,
+        "_op_type": "index",
+    }
+
+
+def create_mapping_document(
+    flow_id: str, flow_name: str, node_id: str, node_title: str
+):
+    """
+    Creates a document containing the flow id to flow title and node id to node title mappings.
+    """
+    return {
+        "flowId": flow_id,
+        "flowName": flow_name,
+        "nodeId": node_id,
+        "nodeTitle": node_title,
+    }
+
+
+def create_document_id(flow_id: str, node_id: str) -> str:
+    """
+    Creates a document id, which depends on a flow and node id.
+    """
+    document_id = f"{flow_id}-{node_id}".encode()
+    return sha256(document_id).hexdigest()
+
+
+def create_mappings(flows: list):
+    """
+    Takes in a set of flows and adds the mappings to bulk index documents, returning the set of
+    bulk actions.
+    """
+
+    bulk_actions = []
+
+    # Going through the flows and pulling out the flow id, flow name, as well as the node id and
+    # node title for all nodes in the flow.
+    for flow in flows:
+        flow_name = flow["name"]
+        flow_id = flow["flowId"]
+        nodes = flow["graphData"]["elements"]["nodes"]
+        for node in nodes:
+            node_id = node["id"]
+            try:
+                # the node title is defined
+                node_title = node["data"]["properties"]["nodeTitle"]["value"]
+            except KeyError:
+                # the node title has not yet been defined
+                node_title = None
+
+            if node_title:
+                # Only add the node id to title mapping if the title is defined for the node
+                document = create_mapping_document(
+                    flow_id, flow_name, node_id, node_title
+                )
+                document_id = create_document_id(flow_id, node_id)
+                # the index the data is added to depends on the current date
+                index_to_update = \
+                  f'{ANALYTICS_INDEX_PATTERN}-{datetime.now().date().strftime("%Y.%m.%d")}'
+                bulk_action = create_bulk_index_action(
+                    index_to_update, document_id, document
+                )
+                bulk_actions.append(bulk_action)
```

Comment:
Check for special case first, then handle it. That way business logic stays mostly flat
```suggestion
            if not node_title:
                continue

            # Only add the node id to title mapping if the title is defined for the node
            document = create_mapping_document(
                flow_id, flow_name, node_id, node_title
            )
            document_id = create_document_id(flow_id, node_id)
            # the index the data is added to depends on the current date
            index_to_update = \
              f'{ANALYTICS_INDEX_PATTERN}-{datetime.now().date().strftime("%Y.%m.%d")}'
            bulk_action = create_bulk_index_action(
                index_to_update, document_id, document
            )
            bulk_actions.append(bulk_action)
```